app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt

```diff
@@ -131,399 +115,170 @@ class ProfileChooserFragmentTest {
-      )
-      onView(withId(R.id.profile_recycler_view)).perform(
-        scrollToPosition<RecyclerView.ViewHolder>(
-          1
-        )
-      )
-      onView(
-        atPositionOnView(
-          R.id.profile_recycler_view,
-          1, R.id.profile_name_text
-        )
-      ).check(
-        matches(
-          withText("Ben")
-        )
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 0)
+      matchStringOnProfileListItem(
+        position = 0,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "Admin"
+      )
+      matchStringOnProfileListItem(
+        position = 0,
+        targetView = R.id.profile_is_admin_text,
+        stringToMatch = context.getString(R.string.profile_chooser_admin)
+      )
+      scrollToPosition(position = 1)
+      matchStringOnProfileListItem(
+        position = 1,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "Ben"
       )
       onView(
         atPositionOnView(
           R.id.profile_recycler_view,
-          1, R.id.profile_is_admin_text
-        )
-      ).check(
-        matches(not(isDisplayed()))
-      )
-      onView(withId(R.id.profile_recycler_view)).perform(
-        scrollToPosition<RecyclerView.ViewHolder>(
-          3
-        )
-      )
-      onView(
-        atPositionOnView(
-          R.id.profile_recycler_view,
-          3, R.id.add_profile_text
-        )
-      ).check(
-        matches(
-          withText(context.getString(R.string.profile_chooser_add))
+          1,
+          R.id.profile_is_admin_text
         )
+      ).check(matches(not(isDisplayed())))
+      scrollToPosition(position = 3)
+      matchStringOnProfileListItem(
+        position = 3,
+        targetView = R.id.add_profile_text,
+        stringToMatch = context.getString(R.string.profile_chooser_add)
       )
     }
   }
 
   @Test
-  // TODO(#973): Fix ProfileChooserFragmentTest
-  @Ignore
-  fun testProfileChooserFragment_initializeProfiles_checkProfilesLastVisitedTimeIsShown() {
-    profileTestHelper.initializeProfiles()
+  fun testProfileChooserFragment_afterVisitingHomeActivity_showsJustNowText() {
+    val data = profileTestHelper.initializeProfiles()
     launch<ProfileChooserActivity>(createProfileChooserActivityIntent()).use {
-      onView(atPosition(R.id.profile_recycler_view, 0)).perform(click())
-      intended(hasComponent(PinPasswordActivity::class.java.name))
-      onView(withId(R.id.input_pin)).perform(typeText("12345"))
-      intended(hasComponent(HomeActivity::class.java.name))
-      onView(isRoot()).perform(pressBack())
-      onView(withText(R.string.home_activity_back_dialog_exit)).perform(click())
-      intended(hasComponent(ProfileChooserActivity::class.java.name))
-      onView(
-        atPositionOnView(
-          R.id.profile_recycler_view,
-          0, R.id.profile_last_visited
-        )
-      ).check(
-        matches(
-          isDisplayed()
-        )
-      )
+      it.onActivity {
```

Comment:
I think we really need a comment here to specify that doing this observer is simulating visiting the home activity, and how.

Reply:
Done